### ISSUE TITLE
Android TV (Nvidia ShieldTV)

### ISSUE THREAD

**Issue by bragg2012** [woodenshark/prismatik-for-android#61]
Running on rooted Android TV (ShieldTV) mood light works but the screen capture doesn't.

Error in the supersu log:
exec /data/data/com.prismadroid.unity/screencaptsrv_api17_armeabi
exit

error: only position independent executables (PIE) are supported

**Comment by bragg2012** [comment]
Any update on this?  Do you need me to get any additional logs or anything to assist?  Happy to help so just let me know if you need anything.

**Comment by e-yes** [comment]
We're working on new implementation of screen capture using new Android5+ API (http://developer.android.com/intl/ru/reference/android/media/projection/MediaProjection.html).
By using this API we don't need this external executable (and mandatory "root" as well).
I expect we release update by end of this week.

Thank you for your patience.

**Comment by bragg2012** [comment]
any update on the release?  Happy to assist with testing if needed.

**Comment by er1kr1984** [comment]
bragg2012 have you had any further update ? is the issue you reported on 0.50

**Comment by bragg2012** [comment]
No further updates received.  I am using Prismatik 0.50 and Android TV version 2.1 which has the issue.  Hopefully a new version will be released that uses the new Android 5+ API soon!

**Comment by HughZurname** [comment]
Happy to assist with testing of this as well.

**Comment by beegmouse** [comment]
+1 on the interest of this

**Comment by bragg2012** [comment]
any update @selfsames ?

**Comment by melstereit** [comment]
+1

**Comment by Dod9er** [comment]
Shield with marshmallow is a superb platform. Getting it to work there would absolutely fabolous :dancer: 
+1

**Comment by bragg2012** [comment]
Are we there yet?

**Comment by beegmouse** [comment]
feedback appreciated

**Comment by Irene-woodenshark** [comment]
Hi, guys!

Tomorrow (23 of March) we will be ready to send a new build (pre-alpha version) of Prismatik for Android 5+ to some of you to test it. Also, we will send some questions to you about the tests. The test will take approximately 4 hours. We will be happy if you can send us a test feedback before the 28th of March. Who is ready to help us?

**Comment by er1kr1984** [comment]
I can help to test the new app.

**Comment by melstereit** [comment]
i want to help with this.

**Comment by beegmouse** [comment]
I'm interested

**Comment by beegmouse** [comment]
Will this new version require root? Or is it using the MediaProjection protocall

I'm not sure I'll have tome to root the shield again between now and the 28th. OS updates undo the root.

**Comment by Irene-woodenshark** [comment]
@beegmouse no, you don't need root it. It uses MediaProjection

**Comment by Irene-woodenshark** [comment]
@er1kr1984 @enclass @beegmouse Hi, guys! If you are still ready, please, send me any mail to irene@woodenshark.com, and I will send you back files. Thank you!

**Comment by beegmouse** [comment]
Hi there

First bit of feedback for you. First the good

Startup animation and moodlighting function correctly.
Lightpack detected.
Zone map is correct, gamma is correct.

Now the bad

Screen capture appears to be trying. as lights turn on and change colour, but colours do not seem to match anything on screen. Top left seems permenantly red, and right and bottom change shades of blue/green. even when there is a static white screen.

Android TV controls are non funtional. CEC, Shield controller, ~~Shield remote~~ cannot navigate app. Edit(Shield remote works, not the controller)

I am still warned of a lack of root at the main screen.

Coloured Radar animation in settings doesn't turn on any lights when testing zones. (This feature has never worked for me, even on ouya) Probably worth removing it. as the zone setup works fine triggering one light at a time.

~~I'll try and get catlog installed when I can, but it won't be until next week.~~
Catlog installed but requires root. I'm not sure how much more help I can be on this.

**Comment by beegmouse** [comment]
A note for you. There is a lot of support from Nvidia for the shield, including debugging and log tools.
https://developer.nvidia.com/develop4shield
It might be worth getting in contact.

**Comment by beegmouse** [comment]
Ok, I got the thing rooted. took forever.
Attached are the recurring erros in catlog

![catlog_20160413_155102](https://cloud.githubusercontent.com/assets/13505070/14497959/960c796a-0190-11e6-84fe-85c16f583f29.png)
![catlog_20160413_155115](https://cloud.githubusercontent.com/assets/13505070/14497971/a361df88-0190-11e6-8593-c5fabf55025d.png)

**Comment by beegmouse** [comment]
Not sure about whats causing the error itself, but it looks like the H value in all the zones is way too big.

**Comment by billymcintosh** [comment]
I'm rooted and ready to test.   How about uploading the code so we can create pull requests?

**Comment by billymcintosh** [comment]
Anyone else willing to share the apk?  

 'woodenshark' has gone MIA again.   Emails go unanswered.  

Would be happy to contribute if given some code.

**Comment by bragg2012** [comment]
Have you tried emailing Irina at irene@woodenshark.com for the apk?

**Comment by billymcintosh** [comment]
Yes, I have.

**Comment by Irene-woodenshark** [comment]
@billymcintosh try to write me once more, please. I do not see your mail in my inbox

**Comment by billymcintosh** [comment]
Thanks, Irene.

Have emailed again.

**Comment by Irene-woodenshark** [comment]
@billymcintosh I see, replied

**Comment by billymcintosh** [comment]
Thanks again.  I'll send logs in due course.

**Comment by bragg2012** [comment]
Does anyone have some instructions/steps to follow for the best way to install the beta apk?

I'm having trouble installing it through 'es file explorer' on the shield TV.

Thanks

**Comment by beegmouse** [comment]
Uninstall any older version of prizmatic
Copy the apk to thumb stick from a PC
Plug it into the shield
Check settings to make sure installing from unkown sources is allowed
Open ES explorer
find and install the apk
Go to settings > apps
Find prizmatic and run

Lightpack need to alter the program to enable a shortcut on android TVs homepage.

**Comment by ant0spy0binns** [comment]
I'm assuming this project is dead

**Comment by beegmouse** [comment]
Slow and unresponsive but not dead.

Irene-woodenshark represents lightpack.tv and only posted 4 days ago.

**Comment by d8ahazard** [comment]
Hello, looking for the beta APK.  I have a shield TV and Nexus Player and can test with both.  I've also emailed Irene directly.  Thanks!

**Comment by henkim21** [comment]
Update on this? I have a shield TV and nexus player also.

**Comment by beegmouse** [comment]
They have just announced Lightpack 2, doing HDMI sampling rather than relying on software.
I think support for the original is all but dead. There is a v0.73 apk of prismatik on lightpack.tv now, so I'll give it a try.

I suspect it is the same we have already been e-mailed though.

**Comment by bragg2012** [comment]
Yeah, just tried 0.73 and it works much better on the Shield, but the
reaction time and the colour matching doesn't seem to be as good as when I
had this configured on kodi/openelec with the lightbob plugin.

The Lightpack 2 might be dead in the water, no specs yet but if it doesn't
support HDMI 2.0 for 4k or HD audio pass through it will be a non-starter.

**Comment by beegmouse** [comment]
Awesome, I'll try 0.73 tonight, why didn't they mention it here.

Suspicious they didn't mention the HDMI rev or HDCP in the teaser video, but there is a big market for 1080 still, bigger than android anyway.I'll keep my fingers crossed

I don't suppose you want to share how you got lightbob plugin working, even rooted mine could never find the boblight daemon? maybe over on the Nvidia forums, as it will get moderated here.

**Comment by bragg2012** [comment]
I used lightbob on an older piece of hardware that was running openelec
which was then replaced with the ShieldTV so haven't actually got it
working on Android.  The lightbob plugin within Kodi was much easier to
configure and much more accurate at capturing the colours than the
prismatik app in my opinion.

**Comment by beegmouse** [comment]
Ah, ok plus it wouldn't have helped with netflix/iplayer/amazonprime/gamestream

fingers crossed.

**Comment by beegmouse** [comment]
Well v0.73 does now function, but it's still unusable.

In menus there is about a second lag, in Kodi 10 seconds, in Gamestream 20-30 seconds and netflix makes the lights go dark.

The app seems to go non responsive when screen capturing is enabled.

**Comment by bragg2012** [comment]
yeah increasing response times would be ideal

**Comment by beegmouse** [comment]
You mean decreasing response times ;-P

**Comment by Jamesh12** [comment]
Amy updates on this really want to get a nvidia shield and a lightpack to work with it cheers guys

**Comment by JohanLeirnes** [comment]
Hi!

I got a Nvidia Shield TV and am wondering if there has been any development on this?
Would be great to get lightpack to work with it.

**Comment by kuroshirokun** [comment]
Hi, I got a Sony Bravia Android TV and two lightpacks. I installed the beta and I can only set the mood light or static light on one of the lightpacks, I didn't manage to make the two work together. Also when I play a movie on plex or kodi the lightpack seems to work but I can't tell for sure because the screen is black.

**Comment by Sim7b** [comment]
Hi, the method with a raspberry pi works ? With a addon for kodi

**Comment by kuroshirokun** [comment]
I updated the firmware today and tried again. Movie is playing I can hear the audio, lights are working but the screen is black

**Comment by JohanLeirnes** [comment]
there seem to be littele to no development on this :(

**Comment by Raphdeumax** [comment]
I have you news on this thread ?
I search to capture all screen apps on android tv but on youtube the mediaprojection return a black bitmap in OnImageAvalaible listener !
Have you an idea ?
Thx
